packages/react/emotion-utils/src/flex.tsx

```diff
@@ -41,19 +41,20 @@ export function flex(
 
 flex.center = (direction?: FlexOptions['direction']) => flex({ justify: 'center', align: 'center', direction });
 
-interface FlexProps<T extends keyof JSX.IntrinsicElements = 'div'> extends AsProps<T>, FlexOptions {}
+export type FlexProps<T extends StringElementType = StringElementType> = AsProps<T> & FlexOptions;
 
-type BaseFlexType = <T extends keyof JSX.IntrinsicElements = 'div'>(
-  props: FlexProps<T> & { ref?: Ref<InferenceHTMLElement<T>> }
-) => ReactElement | null;
+export type BaseFlexType = <T extends StringElementType = StringElementType>(
```

Comment:
I think it would be better to name it `FlexComponentType`.

Reply:
Thanks, cool renaming 👍
There is no Base in Flex so FlexComponentType is much better naming I can understand easily too.